broadinstitute/pearl
# AR-175 OurHealth emails

This adds reminder emails to the platform, and also specifically configures the OurHealth emails.  In the process, it substantially refactors the email service so it can support both individual async emails, and bulk synchronous emails.

This introduces ShedLock to avoid duplicate sending -- we only want one node at a time to be attempting to send reminder emails.

TO TEST:
1. restart AdminApiApp and ParticipantApiApp
2. in line 59 of  `populate/src/main/resources/seed/portals/ourhealth/studies/ourheart/study.json` add property `"afterMinutesIncomplete": 0,` so that reminders will be sent after 0 minutes, rather than the default of 3 days.
3. repopulate ourhealth
4. go to https://localhost:3000/ourhealth/studies/ourheart/env/sandbox
5. click on each of the "Participant Notifications" and send your self a test email from each.  Confirm it matches the content from https://docs.google.com/document/d/1yDd_BjEiJNY1Gmf7CeV2sO5HtzZnPvhU3CiWkZsEqCU/edit
6. Check your email, you should have received a consent reminder for newbie.
